src/popover/container.tsx

```diff
@@ -47,145 +48,206 @@ export default function PopoverContainer({
   size,
   fixedWidth,
   variant,
+  keepPositionWhenInteracting,
 }: PopoverContainerProps) {
   const bodyRef = useRef<HTMLDivElement | null>(null);
   const contentRef = useRef<HTMLDivElement | null>(null);
   const popoverRef = useRef<HTMLDivElement | null>(null);
   const arrowRef = useRef<HTMLDivElement | null>(null);
+  const internalPositionRef = useRef<InternalPosition | null>(null);
 
   const [popoverStyle, setPopoverStyle] = useState<CSSProperties>(INITIAL_STYLES);
-  const [internalPosition, setInternalPosition] = useState<InternalPosition | null>(null);
   const isRefresh = useVisualRefresh();
 
   // Store the handler in a ref so that it can still be replaced from outside of the listener closure.
   const positionHandlerRef = useRef<() => void>(() => {});
 
+  // Remember when the popover was clicked in order to prevent eventually undesired position recalculations coming from that interaction.
+  const clickedInside = useRef(false);
+
+  // Due to applying max height in some cases, it can take more than one render to have the popover in its final desired size.
+  // This ref allows us to track such series of render cycles triggered by a click inside a popover.
+  const resizingAfterInternalClick = useRef(false);
```

Comment:
What if the resize happens asynchronously? It can be 1ms after a click but then the same issue would occur - the popover will reposition. Not relevant for the upcoming feature but does not sound very reliable.

Do we have a use case where the reposition of the chart popover is actually needed? If not, how about adding a flag `keepPosition` to the internal popover and set it to `true` in all chart popovers. When set, the popover would calculate its placement only once upon the first render, or, if we keep the popover open when navigating between series we can possibly also reposition when trackKey changes.

Reply:
> What if the resize happens asynchronously? It can be 1ms after a click but then the same issue would occur - the popover will reposition. Not relevant for the upcoming feature but does not sound very reliable.

If a resize happens for another reason (e.g, an asynchronous content) unrelated to user interaction, even if it happens to happen almost at the same time, then it should reposition. If it happens due to that interaction, then it shouldn't. But it's true that the relationship between the click and the resize on which we base preventing the reposition is a bit fragile (I'm using `requestAnimationFrame` as a proxy for "when the cycle of renders triggered by this interaction has ended", but I can't find a better way to do exactly this with the React official toolset).

> Do we have a use case where the reposition of the chart popover is actually needed?

The use case that I was trying to keep covered was internal asynchronous content updates that could change the popover position without user interaction. Like in [this page](https://d21d5uik3ws71m.cloudfront.net/components/c8cb419f57999afa860ddd22d9a1ca92c919af18/index.html#/light/popover/scrollable). But I guess that is quite far fetched especially for chart popovers. So I'm switching back to  basically ignore popover resize events altogether in chart popovers, as you describe, which was also pretty much the initial approach of this PR.

> if we keep the popover open when navigating between series we can possibly also reposition when trackKey changes

Yes, this is already taken care of [here](https://github.com/cloudscape-design/components/pull/1756/files#diff-10a52bf3488068cab46d14def970625a0a8a12f9ed574481790843a9205c94a6R175).